dlt/destinations/impl/lancedb/errors.py

```diff
@@ -0,0 +1,29 @@
+from functools import wraps
+from typing import (
+    Any,
+)
+
+from lancedb.exceptions import MissingValueError, MissingColumnError  # type: ignore[import-untyped]
+
+from dlt.common.destination.exceptions import (
+    DestinationUndefinedEntity,
+    DestinationTerminalException,
+)
+from dlt.common.destination.reference import JobClientBase
+from dlt.common.typing import TFun
+
+
+def lancedb_error(f: TFun) -> TFun:
```

Comment:
why can't we use raise_database_error for this like in the other destinations? you'll need to adapt _make_database_exception() for the lancedb needs. Or am I missing something here?

Reply:
LanceDB is similar to qdrant and weaviate and doesn't follow sql  semantics or offer a DB API 2.0 specification. Looking at qdrant and weaviate they also only raise `DestinationException`.

Let me know your thoughts?